jetty-core/jetty-websocket/jetty-websocket-jetty-api/src/main/java/org/eclipse/jetty/websocket/api/Frame.java

```diff
@@ -101,4 +101,126 @@ public String toString()
     boolean isRsv2();
 
     boolean isRsv3();
+
+    /**
+     * The effective opcode of the frame accounting for the CONTINUATION opcode.
+     * If the frame is a CONTINUATION frame for a TEXT message, this will return TEXT.
+     * If the frame is a CONTINUATION frame for a BINARY message, this will return BINARY.
+     * Otherwise, this will return the same opcode as the frame.
+     * @return the effective opcode of the frame.
+     */
+    byte getEffectiveOpCode();
+
+    class Wrapper implements Frame
```

Comment:
Is this necessary?
It is only used in 1 place in tests but exposed as public API.

Reply:
It is public API because people might want to use this if they are dealing with only Frames.
We have the same API in WebSocket core to allow people to do this.

Reply:
@lachlan-roberts I do not understand your comment, is there a typo?

In any case, `Wrapper` is only used by `copy()`, which is only used by a test.
I think we should remove both, we do not want to offer APIs that nobody asked and will need to be maintained.
They were not present in 12.0.x and there is no need for them.